pkg/spec/review.go

```diff
@@ -102,23 +109,32 @@ func (s *Spec) ApplyApprovedReview(approvedReviews *ApprovedSpecReview) {
 			mergedPathItem = MergePathItems(mergedPathItem, pathItem)
 
 			// delete path from learning spec
-			delete(s.LearningSpec.PathItems, path)
+			delete(clonedSpec.LearningSpec.PathItems, path)
 		}
 
 		addPathParamsToPathItem(mergedPathItem, pathItemReview.ParameterizedPath, pathItemReview.Paths)
 
 		// add modified path and merged path item to ApprovedSpec
-		s.ApprovedSpec.PathItems[pathItemReview.ParameterizedPath] = mergedPathItem
+		clonedSpec.ApprovedSpec.PathItems[pathItemReview.ParameterizedPath] = mergedPathItem
 
 		// add the modified path to the path tree
-		isNewPath := s.PathTrie.Insert(pathItemReview.ParameterizedPath, pathItemReview.PathUUID)
+		isNewPath := clonedSpec.PathTrie.Insert(pathItemReview.ParameterizedPath, pathItemReview.PathUUID)
 		if !isNewPath {
 			log.Warnf("Path was updated, a new path should be created in a normal case. path=%v, uuid=%v", pathItemReview.ParameterizedPath, pathItemReview.PathUUID)
 		}
 
 		// populate SecurityDefinitions from the approved merged path item
-		s.ApprovedSpec.SecurityDefinitions = updateSecurityDefinitionsFromPathItem(s.ApprovedSpec.SecurityDefinitions, mergedPathItem)
+		clonedSpec.ApprovedSpec.SecurityDefinitions = updateSecurityDefinitionsFromPathItem(clonedSpec.ApprovedSpec.SecurityDefinitions, mergedPathItem)
 	}
+
+	if _, err := clonedSpec.GenerateOASJson(); err != nil {
+		return fmt.Errorf("failed to generate Open API Spec. %v", err)
+	}
+	s.ApprovedSpec = clonedSpec.ApprovedSpec
```

Comment:
How can we make sure that if new fields will be added, this will be updated (and also Clone())? Maybe a comment in the Spec struct?

Reply:
I think we should create a function called "Commit" or something similar that will copy the specific fields from the cloned to the current.
OR just create an internal `spec` struct that will hold all the values except the lock and once verification over just override the internal `spec` struct.
This will ensure any new field/change will be copied.
WDYT?

Reply:
Good point. I will add a comment in the spec. Hopefully the unit tests will catch this as well...
In clone there is no such problem because we marshal and unmarshal the all spec

Reply:
@FrimIdan Not sure I understand your suggestion. How will a Commit function will solve the problem? Also how will an internal spec will solve the problem? still need to make sure we copy everything that needed from the internal spec to the real spec

Reply:
not sure comment is enough

Reply:
for example
```
type internalSpec struct {
	// Host of the spec
	Host string

	Port string
	// Spec ID
	ID uuid.UUID
	// Provided Spec
	ProvidedSpec *ProvidedSpec
	// Merged & approved state (can be generated into spec YAML)
	ApprovedSpec *ApprovedSpec
	// Upon learning, this will be updated (not the ApprovedSpec field)
	LearningSpec *LearningSpec

	PathTrie pathtrie.PathTrie
}

type Spec struct {
	internalSpec

	lock sync.Mutex
}
```

then the copy will be in a single line and no need to handle any new fields
```
s.internalSpec = clonedSpec.internalSpec
```


Not sure `internalSpec` is the right name for it but you get my point.

A `Commit` function was just an idea to arrange all the fields that need to be copied in one place - but the `internalSpec` is better I think

Reply:
Added SpecInfo struct